collector/diskstats_linux.go

```diff
@@ -366,6 +373,19 @@ func (c *diskstatsCollector) Update(ch chan<- prometheus.Metric) error {
 				}
 			}
 		}
+
+		queueStats, err := c.fs.SysBlockDeviceQueueStats(dev)
+		if err != nil {
+			level.Error(c.logger).Log("msg", "Failed to get device queue stats", "err", err)
+			continue
+		}
+		size, err := c.fs.SysBlockDeviceSize(dev)
+		if err != nil {
+			level.Error(c.logger).Log("msg", "Failed to get device size", "err", err)
+			continue
+		}
+		sizeBytes := size * queueStats.LogicalBlockSize
```

Comment:
As I wrote in https://github.com/prometheus/procfs/pull/650#discussion_r1666948030, this will be incorrect for Advanced Format (AF / 4K sector) drives, since `/sys/block/<dev>/size`, which your proposed `SysBlockDeviceSize` method returns, is always reported in units of 512-byte sectors.

Reply:
updated that PR, as well as this one to use the new method 

https://github.com/prometheus/node_exporter/pull/3068/commits/d8e74d65fdada6219fab08383abfb10244cc6858

Reply:
happy with the change? wondering if we can try get these PRs moving

Reply:
LGTM but obviously this is contingent on the procfs PR being accepted first.